text/0000-programmatic-style-injection.md

```diff
@@ -0,0 +1,158 @@
+
+However, until now, the compile-time behavior and runtime behavior for stylesheet modules have never been fully specified. This section fills this specification gap.
+
+Stylesheets modules are identifiable by their `.css` extension. This kind of module only exposes a single default export. The exported value is an opaque object, in other words, the internal implementation of the stylesheet is not exposed to developers. Keeping the exported value opaque is necessary until all the browser vendors agree on [Constructable Stylesheets](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet) implementation.
+
+```js
+import stylesheetA from './styles-a.css';
+import stylesheetB from './styles-b.css';
+```
+
+#### Comparison with CSS module script
+
+LWC stylesheet modules work similarly to native [CSS module script](https://github.com/WICG/webcomponents/blob/gh-pages/proposals/css-modules-v1-explainer.md). This feature isn't fully standardized yet and is today only implemented in Chromium-based browsers. There are few differences between the two proposals worth calling out:
+- LWC stylesheet modules can be imported using a standard import statement. The LWC compiler infers from the `.css` file extension that the file has to be treated as a CSS module. On the other hand, CSS module scripts require the use of [import assertions](https://github.com/tc39/proposal-import-assertions).
+- LWC stylesheet modules default export is an opaque object. CSS module scripts default export is an instance of [`CSSStyleSheet`](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet).
+
+```js
+// LWC stylesheet modules
+import styles from "./styles.css";
+console.log(styles + '') // [object LWCCSSStyleSheet]
+
+// CSS modules
+import styles from "./styles.css" assert { type: "css" };
+console.log(styles + '') // [object CSSStyleSheet]
+```
+
+#### LWC stylesheet import enforcement
+
+As mentioned above, the LWC compiler implicitly loads the stylesheet associated with each template. Internally, the template compiler generates an import statement with the template file name and the `.css` extension. Since stylesheets are optional, the LWC module resolver currently ignores non-existent stylesheet imports.
+
+This behavior is problematic for explicit stylesheet imports as the compilation doesn't fail if the imported stylesheet can't be resolved. To solve this issue a new `optional` query string can be added to the import identifier to indicate whether or not the LWC compiler should report an error when the file is missing.
+
+```js
+// Required LWC stylesheet import. Fails during compilation when missing.
+import styles from "./styles.css";
+
+// Optional LWC stylesheet import. Resolve to an empty stylesheet if missing.
+import styles from "./styles.css?optional=true";
+```

```

Comment:
Not sure if we should call this out as an implementation detail or not. Potentially there are other ways we could solve this problem in the future. (Fancier imports, for instance, `import styles from "./styles.css" assert { type: "lwc/css" } with { optional: true }`.)